### PR TITLE
friends-app

### PR DESCRIPTION
**Pull request by kirill8210**
### friends-app

[Demo](https://kirill8210.github.io/kottans_friends) |
[Code base](https://github.com/kirill8210/kottans_friends)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Review by artem-trubin — CHANGES_REQUESTED** [review]
Hi! I would like to request some changes without looking at the code yet since you have some strange behaviors in your project. First of all, pressing `all gender` seems to reverse the array of displayed users for some reason. Secondly, if you write a name in the search and then click any of the gender options - it clears out the search input (which it should not do). 

Also, please include your other code files, not only the JS one.

**Comment by kirill8210** [comment]
Made edits. Also added other project files

**Review by artem-trubin — CHANGES_REQUESTED** [review]
Hi, sorry for the long delay.

Nice work, but there's a lot to fix and improve here. I'm not going to do a very thorough review, rather guide your thinking to another path and we'll work it out from there. Please follow comments in the code. 

Besides, found a couple of bugs/rough edges in your work, make sure to fix them too:
1. You can only search by users' first names in your work. Last names should be included too.
2. In mobile version, I believe you set your menu to hide every time users changes something. This isn't a good place for such behaviour because:
    1. This isn't a navigation menu where you would like to close it after, for example, navigating to another page. This is a complicated "settings"-like menu, where user possibly would like to change several inputs at once before turning to content.
    2. This behaviour introduced a bug, where the menu would close each time you type a button in the search input, what makes it practically unusable.

Also, please include all code files that you worked with, not only js. html and  css could still contain some mistakes and potential room for improvements.

File: submissions/kirill8210/friends-app/getData.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+let URL_API = 'https://randomuser.me/api/?results=50&nat=us,gb,ua&inc=gender,name,email,dob,phone,picture';
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/getData.js:1]: corrected

File: submissions/kirill8210/friends-app/index.js [hunk 2]
```diff
@@ -0,0 +1,276 @@
+import getData from './getData.js';
+import removeMenu from './mobile.js';
+
+const app = document.querySelector('.main');
+
+let data = [];
+let response = [];
+
+const createCard = (data2) => {
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:9]: corrected

File: submissions/kirill8210/friends-app/index.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    const { name, gender, email, phone, picture, dob } = data2;
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:10]: corrected

File: submissions/kirill8210/friends-app/index.js [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+const sortAge = document.querySelector('.sort_by_age');
+const sortListAge = document.querySelector('.sorts_list_age');
+const sortIdAge = document.querySelector('#sort_by_age');
+const sortName = document.querySelector('.sort_by_name');
+const sortListName = document.querySelector('.sorts_list_name');
+const sortIdName = document.querySelector('#sort_by_name');
+const filterIdGender = document.querySelector('#filters_gender');
+const inputSearch = document.querySelector('#search_friends');
+const reset = document.querySelector('.reset_button');
+const inputAll = document.querySelector('#all');
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:56]: did not understand what to do. move to a separate file, into a function and export the names of the constants?

File: submissions/kirill8210/friends-app/index.js [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+        renderCards(newData);
+        setTimeout(() => {
+            removeMenu();
+        }, 500);
+    }
+});
+
+const sortByAge = () => {
+    if (inputSearch.value.length !== 0) {
+        const fByG = filterByGender()
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:122]: corrected

File: submissions/kirill8210/friends-app/index.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+sortName.addEventListener('click', () => {
+    sortListName.classList.toggle('sorts_list_active');
+    sortListAge.classList.remove('sorts_list_active');
+});
+
+sortListName.addEventListener('click', ({target}) => {
+    if (target.classList.contains('sorts_options')) {
+        sortName.textContent = target.textContent;
+        sortIdName.value = target.dataset.sort;
+        console.log(sortIdName.value);
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:93]: corrected

File: submissions/kirill8210/friends-app/index.js [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+    sortName.textContent = 'Sort by Name';
+    sortIdName.value = 'date';
+    genders = 'all';
+    filterByGender();
+    inputAll.checked = true;
+    inputSearch.value = '';
+    renderCards(data)
+};
+
+reset.addEventListener('click', e => {
```
[the remark this answers is not in the document]
Comment (kirill8210) [reply submissions/kirill8210/friends-app/index.js:251]: honored Google, applied

**Comment by kirill8210** [comment]
fixed bugs and comments

**Comment by kirill8210** [comment]
error correction

**Pull request merged**